agrirouter-sdk-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/encoding/EncodeMessageServiceImpl.java

```diff
@@ -29,26 +50,112 @@ public String encode(
 
     try (ByteArrayOutputStream streamedMessage = new ByteArrayOutputStream()) {
 
-      this.getNativeLogger().trace("Encode header.");
-      this.header(messageHeaderParameters).writeDelimitedTo(streamedMessage);
+      getNativeLogger().trace("Encode header.");
+      header(messageHeaderParameters).writeDelimitedTo(streamedMessage);
 
-      this.getNativeLogger().trace("Encode payload.");
-      this.payload(payloadParameters).writeDelimitedTo(streamedMessage);
+      getNativeLogger().trace("Encode payload.");
+      payload(payloadParameters).writeDelimitedTo(streamedMessage);
 
-      this.getNativeLogger().trace("Encoding message.");
+      getNativeLogger().trace("Encoding message.");
       String encodedMessage = Base64.getEncoder().encodeToString(streamedMessage.toByteArray());
 
-      this.logMethodEnd(encodedMessage);
+      logMethodEnd(encodedMessage);
       return encodedMessage;
     } catch (IOException e) {
       throw new CouldNotEncodeMessageException(e);
     }
   }
 
+  /**
+   * Encode a number of messages.
+   *
+   * @param messageParameterTuples -
+   * @return -
+   */
+  public List<String> encode(List<MessageParameterTuple> messageParameterTuples) {
+    return messageParameterTuples.stream()
+        .map(
+            messageParameterTuple ->
+                encode(
+                    messageParameterTuple.getMessageHeaderParameters(),
+                    messageParameterTuple.getPayloadParameters()))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Chunk a message if necessary. The chunk information and all IDs will be set by the SDK and are
+   * not longer in control of the application.
+   *
+   * @param messageHeaderParameters -
+   * @param payloadParameters -
+   * @return -
+   */
+  public List<MessageParameterTuple> chunk(
+      MessageHeaderParameters messageHeaderParameters,
+      PayloadParameters payloadParameters,
+      OnboardingResponse onboardingResponse) {
+    logMethodBegin(messageHeaderParameters, payloadParameters);
+
+    if (null == messageHeaderParameters || null == payloadParameters) {
+      throw new IllegalArgumentException("Parameters cannot be NULL");
+    }
+    messageHeaderParameters.validate();
+    payloadParameters.validate();
+
+    if (null != messageHeaderParameters.getTechnicalMessageType()
+        && messageHeaderParameters.getTechnicalMessageType().getNeedsChunking()
+        && payloadParameters.shouldBeChunked()) {
+      getNativeLogger()
+          .debug(
+              "The message should be chunked, current size of the payload ({}) is above the limitation.",
+              payloadParameters.getValue().toStringUtf8().length());
+      String wholeMessage = payloadParameters.getValue().toStringUtf8();
```

Comment:
There's a Base64-Encoding issue here. 

Due to the definition, (see https://github.com/DKE-Data/agrirouter-interface-documentation/blob/develop/docs/integration/build-message.adoc#chunking-big-messages), we state that "if the Payload is Base64-Encoded, each chunk needs to be a valid Base64-String".  

If Payload is not Base64-Encoded when it enters this function (as it is in the Test; see corresponding comment), there will not be any Base64-Encoding of the Dataset.

If Payload is Base64-Encoded (which would fit the api definition), we would split a Base64 String in multiple Base64 Chunks. Depending on the length of each Chunk, it can be either valid (33%; no final "=" required) or broken (should end with "=" or "==" but doesn't).

 So, I see 3 possible fixes: 
1.) Base64Decode the Payload and Base64Encode each chunk
2.) Depending on the length of the chunk, add a "=" or "=="
3.) State in the Command parameters, that the Payload shall not be Base64-Encoded AND Base64-Encode the chunks.